File: test/language/expressions/java/expressions/FullConstTypeTest.java
package expressions;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import expressions.full_const_type.FullConstTypeExpression;

public class FullConstTypeTest
{
    @Test
    public void bitSizeOfWithOptional()
    {
        final FullConstTypeExpression fullConstTypeExpression =
                new FullConstTypeExpression(FULL_VALID_VALUE, FULL_ADDITIONAL_VALUE);

        assertEquals(FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, fullConstTypeExpression.bitSizeOf());
    }

    @Test
    public void bitSizeOfWithoutOptional()
    {
        final FullConstTypeExpression fullConstTypeExpression = new FullConstTypeExpression();
        fullConstTypeExpression.setValue(FULL_INVALID_VALUE);

        assertEquals(FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL, fullConstTypeExpression.bitSizeOf());
    }

    private static final int FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 10;
    private static final int FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 7;

    private static final byte FULL_VALID_VALUE = 0x01;
    private static final byte FULL_INVALID_VALUE = 0x00;

    private static final byte FULL_ADDITIONAL_VALUE = 0x03;
}

File: test/extensions/language/expressions/python/FullConstTypeTest.py
import Expressions


class FullConstTypeTest(Expressions.TestCase):
    def testBitSizeOfWithOptional(self):
        fullConstTypeExpression = self.api.FullConstTypeExpression(
            self.FULL_VALID_VALUE, self.FULL_ADDITIONAL_VALUE
        )
        self.assertEqual(
            self.FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, fullConstTypeExpression.bitsizeof()
        )

    def testBitSizeOfWithoutOptional(self):
        fullConstTypeExpression = self.api.FullConstTypeExpression()
        fullConstTypeExpression.value = self.FULL_INVALID_VALUE
        self.assertEqual(
            self.FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL, fullConstTypeExpression.bitsizeof()
        )

    FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 10
    FULL_CONST_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 7

    FULL_VALID_VALUE = 0x01
    FULL_INVALID_VALUE = 0x00

    FULL_ADDITIONAL_VALUE = 0x03
